File: repaso objetos/python/Phyton_Banda/Banda.py
from Musico import Musico
from Clases.Instrumento import *
import random

class Banda():
    
    def __init__(self):
        self.musicos=list()

    def agregarMusico(self,nombre):
        m=Musico()
        m.set__nombre__(nombre)
        self.musicos.append(m)  

    def generarInstrumento(self):
        rn = random.randint(0,3)
        sel = rn
        i=Instrumento()
        if sel == 1:
            i= Guitarra()
            return i
        elif sel == 2:
            i= Bajo()
            return i
        else:
            i= Violin()
            return i
        return i   
    def generarRandom(self):
        a= random.randint(0,2) 
        return a  

    def presentarBanda(self):
        a=Banda()
        for i in self.musicos: 
            i.presentar()
            rn = random.randint(0,2)
            sel = a.generarRandom()
            if sel == 1:
                j= Guitarra()
            elif sel == 2:
                j= Bajo()         
            else:
                j= Violin() 
            print(sel)
            
            
            i.tocar(j)
            
            
               

File: repaso objetos/java/Conciertopaf/Fabrica_especial2/Banda.java
package Conciertopaf.Fabrica_especial2;
import java.util.ArrayList;
import Conciertopaf.Fabrica.Instrumento;
import Conciertopaf.Factory.AireInstrumentFactory;
import Conciertopaf.Factory.CuerdaInstrumentFactory;
import Conciertopaf.Factory.RandomInstrumentFactory;

public class Banda {
    ArrayList<Musico> musicos = new ArrayList<Musico>();

    public void agregarMusico(String nombre){
        Musico m=new Musico();
        m.setNombre(nombre);
        musicos.add(m);
    }

    private Instrumento generarInstrumento(int tip){
        switch(tip){
            case 0:
                return new RandomInstrumentFactory().seleccionar();
            case 1:
                return new AireInstrumentFactory().seleccionar();
            default:
                return new CuerdaInstrumentFactory().seleccionar();
        }
    }

    public void presentarBanda(int tip){
        for (Musico musico : musicos){
            musico.presentar();
            
            musico.tocar(generarInstrumento(tip));
        }
    }
}

